ssm/lds.py

```diff
@@ -609,52 +659,129 @@ def hessian_neg_expected_log_joint(x, Ez, Ezzp1, scale=1):
 
             # Use Newton's method or LBFGS to find the argmax of the expected log joint
             scale = x0.size
-            obj = lambda x: neg_expected_log_joint(x, Ez, Ezzp1, scale=scale)
+
+            def obj(x):
```

Comment:
Why define both `obj` and `_objective`?

Reply:
I was mostly just copying what was in master -- it seems like maybe scipy.optimize.minimize expects an objective function that takes in two arguments (x, iter) whereas the block-tridiagonal newton method expects an objective that only takes in x? 

I'm looking at optimizers.py here https://github.com/slinderman/ssm/blob/82641a1930923c4cd96d1c073c407525a93ed8d8/ssm/optimizers.py#L122

it seems like we pass in -1 as the iteration, but I'm not sure why.

Reply:
Yea, it's because the ssm.optimizers functions take in an objective and an iteration count (following the autograd model).  That's so that you can have iteration-dependent step sizes, e.g.  I think we can define objective once and then pass a lambda function with an extra `itr` arg for the lbfgs case.

Reply:
Oh the dependent step-sizes makes sense. Jax has a pretty clean way to do this with their optimizers package. I'll clean this up as you mentioned.